dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/data/upload/DataUploadRunnableTest.kt

```diff
@@ -121,60 +123,97 @@ internal class DataUploadRunnableTest {
         )
     }
 
-    @ParameterizedTest
-    @EnumSource(
-        SystemInfo.BatteryStatus::class,
-        names = ["DISCHARGING", "NOT_CHARGING"],
-        mode = EnumSource.Mode.INCLUDE
-    )
-    fun `doesn't send batch when battery is low and unplugged`(
-        batteryStatus: SystemInfo.BatteryStatus,
+    @Test
+    fun `M send batch W run() { batteryFullOrCharging }`(@Forgery batch: Batch) {
+        val fakeSystemInfo = SystemInfo(batteryFullOrCharging = true)
+        whenever(mockReader.lockAndReadNext()) doReturn batch
+        whenever(mockSystemInfoProvider.getLatestSystemInfo()) doReturn fakeSystemInfo
+        whenever(mockDataUploader.upload(batch.data)) doReturn UploadStatus.SUCCESS
+
+        testedRunnable.run()
+
+        verify(mockReader).drop(batch)
+        verify(mockReader, never()).release(batch)
+        verify(mockDataUploader).upload(batch.data)
+        verify(mockThreadPoolExecutor).schedule(
+            same(testedRunnable),
+            any(),
+            eq(TimeUnit.MILLISECONDS)
+        )
+    }
+
+    @Test
+    fun `M send batch W run() { battery level high }`(
         @Forgery batch: Batch,
-        forge: Forge
+        @IntForgery(min = DataUploadRunnable.LOW_BATTERY_THRESHOLD + 1) batteryLevel: Int
     ) {
-        val systemInfo = SystemInfo(
-            batteryStatus,
-            forge.anInt(1, 10)
+        val fakeSystemInfo = SystemInfo(batteryLevel = batteryLevel)
```

Comment:
Can we explicit the other values in system Info:
```
SystemInfo(
    batteryLevel = batteryLevel,
    batteryFullOrCharging = randomBool(), 
    onExternalPowerSource = randomBool(),
    powerSaveMode = false
)
```

Reply:
same remark as above, you will introduce false positives